### DISCUSSION TITLE
Reducing initial delay while reading

### DISCUSSION THREAD

**Discussion by nishant-gh** [bluenviron/mediamtx#3061] [Questions]
### Question

My MediaMTX server pulls RTSP streams from cameras. And I am using WebRTC output from the server to show live stream in browser/UI. While this works great for most of the sources, some of the streams suffer from initial delay in the UI (sometimes as long as 30s or more). This could probably be due to network connectivity (I am not sure about this though as we don’t have direct control over the sources and the environment). However when the stream begins to play it continues to play fine more or less. We are not using ‘sourceOnDemand’ so the delay couldn’t possibly be because of stream not being “ready” when a client connects.

I guess my question is - is there any way to get rid of the initial delay as the stream seems to work fine (it is almost real time as expected) after that? Can the stream be “ready” as soon as a client connects?

**Comment by aler9** [comment]
Hello, a player can decode a stream only after it receives an I-frame. Therefore, the parameter you're looking for is the "I-frame interval", that can be tuned on the web interface of your cameras. Decrease it and you will decrease the maximum delay before a stream can be decoded.

[a comment by an automated account is not shown]
